src/entities/speller/ui/speller-setting.tsx

```diff
@@ -41,14 +55,18 @@ const SpellerSetting = memo(() => {
         id='airplane-mode'
         name='isStrictCheck'
         onCheckedChange={handleChange}
-        defaultChecked={Boolean(
-          searchParams?.get(QUERY)?.toString() === 'true',
-        )}
+        checked={isChecked}
```

Comment:
```suggestion
        checked={searchParams?.get(QUERY) === 'true'}
```

[P1] `useSearchParams()` 훅이 항상 새로운 값을 반환하기 때문에 useEffect와 useState를 사용하지 않아도 될 것 같아요!

그리고 `isStrictCheckEnabled`은 해당 파일에서 중복으로 사용하는 부분이 없고, searchParams 여부는 옵셔널 체이닝으로 처리해도 돼서 별도의 변수로 선언하지 않고 그대로 드러내는 것도 좋겠다는 생각이 드는데 정배님 생각은 어떠실까요?

Reply:
@bicochan 
1. 넵 현재 로직에서는 useEffect와 useState는 불필요한것 같네요 수정했습니다!
2. searchParams?.get(QUERY) === 'true'는 의도를 파악하려면 QUERY 값과 비교 로직을 따라가야 하지만, isStrictCheckEnabled()와 같이 함수명으로 추상화하면 코드의 의도가 명확해지고, 구현은 숨길 수 있어 가독성과 유지보수성이 높아진다고 생각되서 로직을 함수로 분리하였습니다.

Reply:
`isStrictCheckEnabled`라는 이름도 좋네요!

그런데 이미 쿼리명이 명확해서 아래와 같이 작성해도 괜찮겠다는 생각도 들었어요(선택은 정배님이...😊).

```
checked={searchParams?.get('isStrictCheck') === 'true'}
```